### PR TITLE
fix: Use DB credentials for Redis Streams MesssageBus connection

### PR DESCRIPTION
**Pull request by lenny-intel**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**If your build fails** due to your commit message not passing the build checks, please review the guidelines here: https://github.com/edgexfoundry/edgex-examples/blob/master/.github/CONTRIBUTING.md.

## What is the current behavior?

Can not use Redis Streams MesssageBus in secure mode due to lack of password on connection.

## What is the new behavior?

Now the DB credentials are used for the Redis Streams MessageBus connection enabling it to work in secure mode.

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

## New Imports
Are there any new imports or modules? If so, what are they used for and why?
no

## Specific Instructions
Are there any specific instructions or things that should be known prior to reviewing?

## Other information

### CODE REVIEW

**Review by jpwhitemn — APPROVED** [review]
LGTM

File: internal/core/data/init.go [hunk 1]
```diff
@@ -62,6 +62,19 @@ func (b *Bootstrap) BootstrapHandler(ctx context.Context, wg *sync.WaitGroup, st
 	mdc := metadata.NewDeviceClient(local.New(configuration.Clients["Metadata"].Url() + clients.ApiDeviceRoute))
 	msc := metadata.NewDeviceServiceClient(local.New(configuration.Clients["Metadata"].Url() + clients.ApiDeviceRoute))
 
+	// For Redis Streams MessageBus, we reuse the Redis instance running for the DB, which may have a password,
+	// so we need to get and use the DB credentials for the MessageBus connection.
+	if configuration.MessageQueue.Type == "redisstreams" {
```
Comment (jim-wang-intel) [inline internal/core/data/init.go:67]: Do case insensitive for this?

**Review by jim-wang-intel — APPROVED** [review]

File: internal/core/data/init.go (hunk 1 above)
Comment (lenny-intel) [reply internal/core/data/init.go:67]: Nope, not consistent with what MessageBus factory is doing.

**Review by andresrinivasan — COMMENTED** [review]
Good for now. We'll need to circle back for Redis 6 ACLs.

**Pull request merged**